agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -475,9 +478,9 @@ bool slave_thread::handle_cmdu_control_message(
     // LOG(DEBUG) << "handle_cmdu_control_message(), INTEL_VS: action=" + std::to_string(beerocks_header->action()) + ", action_op=" + std::to_string(beerocks_header->action_op());
     // LOG(DEBUG) << "received radio_mac=" << network_utils::mac_to_string(beerocks_header->radio_mac()) << ", local radio_mac=" << network_utils::mac_to_string(hostap_params.iface_mac);
 
-    if (!std::equal(beerocks_header->radio_mac().oct,
-                    beerocks_header->radio_mac().oct + net::MAC_ADDR_LEN,
-                    hostap_params.iface_mac.oct)) {
+    // to me or not to me, this is the question...
+    if (network_utils::mac_to_string(beerocks_header->radio_mac()) !=
+        network_utils::mac_to_string(hostap_params.iface_mac)) {
```

Comment:
Agree, I didn't want to do that since it is not related to this change but since you brought it up i'll do the change, in a separate commit.
I have no idea why sMacAddr have channel and rssi - probably someone who was too lazy to define a wrapper struct - I will fix this as well.